File: Java/problem70-2.java
class Solution {
    public int climbStairs(int n) {
        int dp0 = 1, dp1 = 1;
        for (int i = 2; i <= n; ++i) {
            int temp = dp1;
            dp1 = dp0 + dp1;
            dp0 = temp;
        }
        return dp1;
    }
}

File: python/problem70-2.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp0 = 1
        dp1 = 1
        for i in range(2, n + 1):
            temp = dp1
            dp1 = dp0 + dp1
            dp0 = temp
        return dp1
